Initial draft of docs

shannonxtreme opened the pull request:
Added Design Requirements topic and experimented with folder structure and TOC. 
Made changes to the Design Requirements topic and added comments to source file in `Overview/design-requirements.rst`
Added requirements for the site introduction page.

ClarkPoon commented on Docs/source/Overview/design-requirements.rst:
```diff
@@ -0,0 +1,28 @@
+Design requirements vary based on the type of batteries you use. For example, lead-acid packs require less protection than lithium-ion cells. BeVolt’s battery pack uses Samsung INR18650-30Q lithium-ion cells. The BPS is required to monitor the current, voltage, and temperature of the battery pack.
+
+.. 
+    Packs vs cells - do lead-acid batteries only come as single units or can you get lead-acid cells? Can a battery pack contain one of either li-ion cells or lead-acid cells? If so, 
```
Lithium-ion and lead-acid batteries both have cells. Multiple cells are combined to create a pack. I see the issue here where we are comparing lithium-ion **cells** to lead-acid **packs.** The idea we are trying to get across is that a lithium-ion battery is less stable than a lead-acid battery, so you need more protection for lithium-ion than lead-acid.

ClarkPoon commented on Docs/source/Overview/design-requirements.rst:
```diff
@@ -0,0 +1,28 @@
+.. 
+    Packs vs cells - do lead-acid batteries only come as single units or can you get lead-acid cells? Can a battery pack contain one of either li-ion cells or lead-acid cells? If so, 
+    we need to be specific. 
+    Link to cell datasheet would be good. 
```
Cell datasheet is here: https://www.18650batterystore.com/v/files/samsung_30q_data_sheet.pdf

I also have a pdf on our team Google Drive, if that would be better.

ClarkPoon commented on Docs/source/Overview/design-requirements.rst:
```diff
@@ -0,0 +1,28 @@
+    * Trip system that disconnects the battery from the car when:
+        
+        * the voltage of a module exceeds the safe range
+        * the temperature exceeds the maximum specified in the datasheets
```
We also disconnect the battery from the car when the current through the battery exceeds the safe bounds.

ClarkPoon commented on Docs/source/Overview/design-requirements.rst:
```diff
@@ -0,0 +1,28 @@
+        * the voltage of a module exceeds the safe range
+        * the temperature exceeds the maximum specified in the datasheets
+        .. note::
+            The temperature limit varies depending on whether the battery is charging or discharging. 
```
The current limit also depends on if the battery is charging or discharging.

ClarkPoon commented on Docs/source/introduction.rst:
```diff
@@ -0,0 +1,8 @@
+================================
+Provide an introduction to the system. The introduction should cover the following topics in brief: 
+
+    1. What is a BPS? 
```
I am guessing this is a list of questions that my team needs to answer for you?

ClarkPoon reviewed (commented):
This looks nice. I added some comments to fill in some missing information.

shannonxtreme commented:
@ClarkPoon I added a new section and implemented your comments. Also added links and a Glossary to link terms to. 

For the image showing the entire system, highly recommend we make the board/component names consistent with the documentation. For example, the Current board is called the Amperes board in the image. If you have a source file for the image I can tweak it!

ClarkPoon commented on Docs/source/Overview/design-overview.rst:
```diff
@@ -0,0 +1,60 @@
+A Display board is fixed on the dashboard. The BPS Display board is separate from the dashboard to keep it independent from all other systems while debugging.
+
+.. 
+  Does this mean there are two Display boards? Or is the one in your car not on the dashboard? 
```
@SijWoo what did you have in mind here?

SijWoo replied:
Sorry for the confusion, there are two displays in the car: dashboard display and BPS display. One is on the dashboard that displays speed and cruise control modes; the other display is the BPS display board which displays the **battery percentage, current flow in amps, total voltage, average temperature, and an indicator if the BPS faulted**. The BPS Display is connected to the main CAN Bus. The BPS Display is powered from the BPS, so whenever a trip happens and the battery pack is disconnected from the electrical system, the BPS display is still powered. The BPS display will be in the cockpit but probably not fixed on the dashboard. The BPS display will be mounted on an arm that can be adjusted.

ClarkPoon commented on Docs/source/Overview/design-overview.rst:
```diff
@@ -0,0 +1,60 @@
+.. note::
+  The minion boards all use `isolated SPI <https://www.analog.com/en/products/interface-isolation/isolation/spisolator.html>`_. 
+.. 
+  There's no flow between the isolated SPI sentence and the following sentence. 
```
This information is outdated, and the minion boards and amperes board (formerly known as Current board) all use isolated SPI, so we can get rid of "The Current board sends two analog signals to the master board"

ClarkPoon commented:
> @ClarkPoon I added a new section and implemented your comments. Also added links and a Glossary to link terms to.
> 
> For the image showing the entire system, highly recommend we make the board/component names consistent with the documentation. For example, the Current board is called the Amperes board in the image. If you have a source file for the image I can tweak it!

Thanks, it looks nice.

I agree 100%. We recently changed the name of the Current board to the Amperes board because we felt that it was confusing on when "current board" meant "board that measures electric current" or "most recent version of board." Unfortunately, we have not had a chance to change the name in all of our documentation yet.

ClarkPoon reviewed (approved):
I recognize that there is still room for improvement here, but I want to merge this into master, so the rest of the BPS team can use this as a guide for how to add to the documentation. What do you think @shannonxtreme @SijWoo @chaseblock ?
